airflow/kubernetes/pod_generator.py

```diff
@@ -298,51 +361,171 @@ def reconcile_pods(base_pod: k8s.V1Pod, client_pod: k8s.V1Pod) -> k8s.V1Pod:
+            - airflow.cfg
+            - executor_config
+            - dynamic arguments
+        """
+
+        dynamic_pod = PodGenerator(
+            namespace=namespace,
+            labels={
+                'airflow-worker': worker_uuid,
+                'dag_id': dag_id,
+                'task_id': task_id,
+                'execution_date': date,
+                'try_number': str(try_number),
+                'airflow_version': airflow_version.replace('+', '-'),
+                'kubernetes_executor': 'True',
+            },
+            cmds=command,
+            name=pod_id
+        ).gen_pod()
+
+        # Reconcile the pod generated by the Operator and the Pod
+        # generated by the .cfg file
+        pod_with_executor_config = PodGenerator.reconcile_pods(worker_config,
+                                                               kube_executor_config)
+        # Reconcile that pod with the dynamic fields.
+        return PodGenerator.reconcile_pods(pod_with_executor_config, dynamic_pod)
+
+    @staticmethod
+    def deserialize_model_file(api_client: ApiClient, path: str) -> k8s.V1Pod:
+        """
+        :param api_client: K8S client object
+        :param path: Path to the file
+        :return: a kubernetes.client.models.V1Pod
+        """
+        pod = None
+        with open(path) as stream:
+            if '.json' in path:
+                pod = json.load(stream)
+            elif '.yaml' in path:
+                pod = yaml.safe_load(stream)
+            elif not pod:
+                raise AirflowConfigException("Path was neither .json nor .yaml")
+
+            # pylint: disable=protected-access
+            return api_client._ApiClient__deserialize_model(pod, k8s.V1Pod)
+
+    @staticmethod
+    def deserialize_model_string(api_client: ApiClient, string: str) -> k8s.V1Pod:
+        """
+        :param api_client: K8S client object
+        :param string: a string of the deployment
+        :return: a kubernetes.client.models.V1Pod
+        """
+        try:
+            pod = json.loads(string)
+        except json.decoder.JSONDecodeError:
+            try:
+                pod = yaml.safe_load(string)
+            except ScannerError:
+                raise AirflowConfigException(
+                    "Could not parse {} as yaml or json".format(string)
+                )
+
+        # pylint: disable=protected-access
+        return api_client._ApiClient__deserialize_model(pod, k8s.V1Pod)
+
+    @staticmethod
+    def deserialize_model(path_or_string) -> k8s.V1Pod:
+        """
+        :param path_or_string: path to a pod JSON/YAML or a string
+        :return: a kubernetes.client.models.V1Pod
+        """
+        api_client = ApiClient()
+        if os.path.exists(path_or_string):
+            return PodGenerator.deserialize_model_file(api_client, path_or_string)
+
+        return PodGenerator.deserialize_model_string(api_client, path_or_string)
```

Comment:
When would this path actually be hit? It's ultimately coming from a config file so the file version is the only one that we should need, right?

Reply:
Argument could be string in the case of the `KubernetesPodOperator` or in the case of the `executor_config` argument to any operator when running the `KubernetesExecutor`.

Reply:
If it's coming from PodOp or `executor_config` wouldn't a config object be better than a JSON/YAML string?

Reply:
Definitely passing the pod object directly is better. I'll remove the option to pass a string.